src/main/java/org/sunbird/common/service/impl/ElasticSearchRestHighImpl.java

```diff
@@ -0,0 +1,172 @@
+                    } else {
+                        Map<String, Object> map = (Map<String, Object>) entry.getValue();
+                        Map<String, String> dataMap = (Map) map.get(Constants.TERM);
+                        for (Map.Entry<String, String> dateMapEntry : dataMap.entrySet()) {
+                            FieldSortBuilder mySort = new FieldSortBuilder(entry.getKey() + ElasticSearchHelper.RAW_APPEND).setNestedFilter(new TermQueryBuilder(dateMapEntry.getKey(), dateMapEntry.getValue())).sortMode(SortMode.MIN).order(ElasticSearchHelper.getSortOrder((String) map.get(Constants.ORDER)));
+                            searchSourceBuilder.sort(mySort);
+                        }
+                    }
+                }
+            }
+
+            // apply the fields filter
+            searchSourceBuilder.fetchSource(searchDTO.getFields() != null ? searchDTO.getFields().stream().toArray(String[]::new) : null, searchDTO.getExcludedFields() != null ? searchDTO.getExcludedFields().stream().toArray(String[]::new) : null);
+
+            // setting the offset
+            if (searchDTO.getOffset() != null) {
+                searchSourceBuilder.from(searchDTO.getOffset());
+            }
+
+            // setting the limit
+            if (searchDTO.getLimit() != null) {
+                searchSourceBuilder.size(searchDTO.getLimit());
+            }
+            // apply additional properties
+            if (searchDTO.getAdditionalProperties() != null && searchDTO.getAdditionalProperties().size() > 0) {
+                for (Map.Entry<String, Object> entry : searchDTO.getAdditionalProperties().entrySet()) {
+                    ElasticSearchHelper.addAdditionalProperties(query, entry, constraintsMap);
+                }
+            }
+
+            //apply multimatch on query
+            if (searchDTO.getMultiSearchFields() != null && searchDTO.getMultiSearchFields().size() > 0) {
+                for (Map.Entry<String, List<String>> entry : searchDTO.getMultiSearchFields().entrySet()) {
+                    query.should(ElasticSearchHelper.createMultiMatchQuery(entry.getKey(), entry.getValue().toArray(new String[entry.getValue().size()]), null));
+                }
+            }
+
+            // set final query to search request builder
+            searchSourceBuilder.query(query);
+            List finalFacetList = new ArrayList();
+
+            if (null != searchDTO.getFacets() && !searchDTO.getFacets().isEmpty()) {
+                searchSourceBuilder = addAggregations(searchSourceBuilder, searchDTO.getFacets());
+            }
+            logger.info("ElasticSearchRestHighImpl:search: calling search for index " + index + ", with query = " + searchSourceBuilder.toString());
+
+            searchRequest.source(searchSourceBuilder);
+            SearchResponse response = esClient.search(searchRequest, RequestOptions.DEFAULT);
```

Comment:
Need to explore the possibility for getting only specific fields from the matching objects instead of entire object. this would avoid us looping through and filtering the fields when sending back response.